set.mm

```diff
@@ -507326,6 +507696,61 @@ codes is an increasing chain (with respect to inclusion).  (Contributed
       NUIUABJKUGUHOPQFERT $.
   $}
 
+  ${
+    $d A a $.  $d B a $.  $d M a $.  $d V a $.
+    $( The wff ` ( A e. B -/\ B e. A ) ` encoded as ` ( ( A e.g B ) `
+       ` |g ( B e.g A ) ) ` is true in any model ` M ` .  This is the model
+       theoretic proof of ~ elnanel .  (Contributed by AV, 5-Nov-2023.) $)
+    elnanelprv $p |- ( ( M e. V /\ A e. _om /\ B e. _om )
+                       -> M |= ( ( A e.g B ) |g ( B e.g A ) ) ) $=
+      ( va wcel com w3a cgoe co cgna cprv wbr csate cmap wceq cfv wn wa cvv a1i
+      cv crab simp1 3simpc pm3.22 3adant1 eqid satefvfmla1 syl3anc wnan elnanel
+      wo nanor mpbi rabeqc syl6eq wb ovex prv sylancl mpbird ) CDFZAGFZBGFZHZCA
+      BIJZBAIJZKJZLMZCVINJZCGOJZPZVFVKAEUBZQZBVNQZFZRVPVOFZRUMZEVLUCZVLVFVCVDVE
+      SVEVDSZVKVTPVCVDVEUDZVCVDVEUEVDVEWAVCVDVEUFUGABBACDVIEVIUHUIUJVSEVLVSVNVL
+      FVQVRUKVSVOVPULVQVRUNUOUAUPUQVFVCVITFVJVMURWBVGVHKUSVICDTUTVAVB $.
+  $}
+
+  ${
+    $d U x $.
+    $( Every wff encoded as ` U ` is true in an empty model.  (Contributed by
+       AV, 19-Nov-2023.) $)
+    prv0 $p |- ( U e. ( Fmla ` _om ) -> (/) |= U ) $=
+      ( vx com cfmla cfv wcel c0 cprv wbr csate co wceq csat sate0 cv wn peano1
+      wa cvv 0ex wal wf n0ii intnan a1i f00 sylnibr pm3.2i satfvel mp3an1 mtand
+      alrimiv eq0 sylibr eqtrd cmap prv mpan wne ne0ii map0b mp1i eqeq2d mpbird
+      wb bitrd ) ACDEZFZGAHIZGAJKZGLZVHVJACGGMKEEZGAVGNVHBOZVLFZPZBUAVLGLVHVOBV
+      HVNCGVMUBZVHVMGLZCGLZRZVPVSPVHVRVQGCQUCUDUECVMUFUGGSFZVTRVHVNVPVTVTTTUHVM
+      AGGSSUIUJUKULBVLUMUNUOVHVIVJGCUPKZLZVKVTVHVIWBVETAGSVGUQURVHWAGVJCGUSWAGL
+      VHGCQUTCVAVBVCVFVD $.
+  $}
+
+  ${
+    $d I a $.  $d J a $.  $d V a $.  $d X a $.
+    $( No wff encoded as a Godel-set of membership is true in a model with only
+       one element.  (Contributed by AV, 19-Nov-2023.) $)
+    prv1n $p |- ( ( I e. _om /\ J e. _om /\ X e. V )
+                  -> -. { X } |= ( I e.g J ) ) $=
```

Comment:
Is my understanding of this theorem correct?
> In `set.mm`, because of `eirrv`, which is derived from the axiom of regularity, one cannot have `A e. A`.
> And because our model/satisfiability theory `Sat` and `|=` is based on `set.mm` ZFC classes, this also cannot hold in any of our models.

But then, what is the use of `AxReg`, defined in `~ df-gzreg`?

The same is obviously valid for all axioms, I used `ax-reg` here because this is the one linked with this theorem.

Reply:
I guess proving theorems like `prv1n` will be pretty hard (not mathematically, but in terms of tracking axiom dependencies) with current tools, because you want to avoid to use all ZF axioms "on the syntactic side", but all intermediate theorems we used to build the machinery of model theory drag with them many axiom dependencies... (but maybe not the axiom of regularity if we were careful?).

Probably, the theorem we want to prove is something like
```
 prv1n $p |- ( ( I e. _om /\ J e. _om /\ X e. V ) ->
       ( -. { X } |= AxReg -> -. { X } |= ( I e.g J ) ) ) $=
```
while avoiding the "genuine" uses of ax-reg

Reply:
> But then, what is the use of `AxReg`, defined in `~ df-gzreg`?
> 
> The same is obviously valid for all axioms, I used `ax-reg` here because this is the one linked with this theorem.

That is a really good question. I also have no idea how the axioms could/would be integrated into the framework of the current definitions. This would be a task to be done in the context of issue #3599.  And this should be done before all the stuff is moved to main.

Reply:
I don't understand: I can see my answer from yesterday above, but maybe it's because I'm logged in, or there is a delay.  I'm copying it below and hopefully it will work this time:

[
I guess proving theorems like prv1n will be pretty hard (not mathematically, but in terms of tracking axiom dependencies) with current tools, because you want to avoid to use all ZF axioms "on the syntactic side", but all intermediate theorems we used to build the machinery of model theory drag with them many axiom dependencies... (but maybe not the axiom of regularity if we were careful?).

Probably, the theorem we want to prove is something like
```
 prv1n $p |- ( ( I e. _om /\ J e. _om /\ X e. V ) ->
       ( -. { X } |= AxReg -> -. { X } |= ( I e.g J ) ) ) $=
```
while avoiding the "genuine" uses of ax-reg
]

What I called "theorems we used to build the machinery of model theory" is what Mario calls the metatheory.

This does not prevent merging of this MR, since this is a longer term work.

Reply:
> This does not prevent merging of this MR, since this is a longer term work.

Yes, I completely agree. 
My point was more about trying to understand what we are proving here.